src/orange/transform/SignedPermutation.hh

```diff
@@ -0,0 +1,199 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/transform/SignedPermutation.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/cont/EnumArray.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/cont/Span.hh"
+#include "corecel/math/Algorithms.hh"
+#include "corecel/math/Turn.hh"
+#include "orange/OrangeTypes.hh"
+#include "orange/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Apply a rotation that remaps and possibly flips signs.
+ *
+ * The daughter-to-parent rotation matrix in this special case:
+ * \f[
+ \mathbf{R} = \begin{bmatrix}
+  \mathbf{e}_x \\ \hline
+  \mathbf{e}_y \\ \hline
+  \mathbf{e}_z
+  \end{bmatrix}
+  \f]
+ * where \f$ \mathbf{e}_u \f$ is has exactly one entry with a value \f$ \pm 1
+ \f$ and the other entries being zero.
+ *
+ * TODO: implement error checking to prevent reflection.
+ *
+ * The underlying storage are a compressed series of bits in little-endian form
+ * that indicate the positions of the nonzero entry followed by the sign:
+ * \verbatim
+   [flip z'][z' axis][flip y'][y' axis][flip x'][x' axis]
+         8   7    6        5   4     3        2   1   0  bit position
+ * \endverbatim
+ * This allows the "rotate up" to simply copy one value at a time into a new
+ * position, and optionally flip the sign of the result.
+ */
+class SignedPermutation
+{
+  public:
+    enum Sign : char
+    {
+        pos = '+',
```

Comment:
In some places in this class, pos means positive, whereas in other places it means position. This actually confused me when looking at the constructor in the .cc. Consider revising.

Reply:
I've renamed to psign/msign, but I think it might just be better to hard-code `'+'` and `'-'` everywhere instead... let me know if you have an opinion. I generally try to avoid hardcoded strings or magic characters, but this might be a good exception.

Reply:
I agree that hardcoded is better in this case, as you decided.